core/src/test/java/jenkins/cli/StopBuildsCommandTest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, Ilia Zasimov
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.cli;
+
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractItem;
+import hudson.model.AbstractProject;
+import hudson.model.Executor;
+import jenkins.model.Jenkins;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.StringContains.containsString;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.*;
+
+@RunWith(PowerMockRunner.class)
```

Comment:
I tend to not trust any tests written with PowerMock. They just prove what I could already tell by looking at the source code. They do not prove that the feature actually works at all. YMMV

Reply:
It's seems like a valid technique for mocking` Jenkins.get()` method. In other tests it works in the same way (`NewViewLinkTest` for example). If you insist to change it, please show me how to inject `Jenkins` instance in `StopBuildsCommand` without `PowerMock`.

Reply:
I know there are other tests using mocks; I do not trust them either! I am not insisting on changing anything (this was a side comment), but if you do want to try a functional test they use `JenkinsRule` and (for CLI commands) `CLICommandInvoker`.

Reply:
Tests now implemented with `JenkinsRule` and `CLICommandInvoker`